RecoPixelVertexing/PixelTriplets/src/CACut.h

```diff
@@ -0,0 +1,169 @@
+    if (tripletCuts.size() == 1 && tripletCuts[0].getParameter<double>("cut") == -1.) {
+      useCACuts = false;
+      //TODO Uncomment the Following warning after tuning
+      //edm::LogWarning("Configuration") << "No CACut VPSet. Using default cut value of " << defaultCut << " for all layer triplets";
+      return;
+    }
+
+    valuesByTripletNames_.reserve(tripletCuts.size());
+    valuesByLayerIds_.reserve(tripletCuts.size());
+
+    setCutValuesByTripletNames(tripletCuts);
+  }
+
+  void setCutValuesByTripletNames(const std::vector<edm::ParameterSet> &tripletCuts) {
+    for (const auto &thisTriplet : tripletCuts) {
+      valuesByTripletNames_.emplace_back();
+      auto &thisCACut = valuesByTripletNames_.back();
+
+      thisCACut.tripletName = thisTriplet.getParameter<std::string>("seedingLayers");
+      thisCACut.cutValue = thisTriplet.getParameter<double>("cut");
+    }
+  }
+
+  void setCutValuesByLayerIds(CAGraph &caLayers) {
+    if (!useCACuts || foundAllLayerIds)
+      return;
+
+    foundAllLayerIds = true;
+    valuesByLayerIds_.clear();
+
+    for (const auto &thisTriplet : valuesByTripletNames_) {
+      valuesByLayerIds_.emplace_back();
+      auto &thisCACut = valuesByLayerIds_.back();
+
+      // Triplet name, e.g. 'BPix1+BPix2+BPix3'
+      std::string layersToSet = thisTriplet.tripletName;
+      for (int thisLayer = 0; thisLayer < 3; thisLayer++) {
+        // Get layer name
+        std::size_t layerPos = layersToSet.find("+");
+        if ((thisLayer < 2 && layerPos == std::string::npos) || (thisLayer == 2 && layerPos != std::string::npos)) {
+          throw cms::Exception("Configuration")
+              << "Please enter a valid triplet name in the CACuts parameter set; e.g. 'BPix1+BPix2+BPix3'";
+        }
+
+        std::string layerName = layersToSet.substr(0, layerPos);
+        layersToSet = layersToSet.substr(layerPos + 1);
+
+        // Get layer ID
+        thisCACut.layerIds.emplace_back(caLayers.getLayerId(layerName));
+        if (thisCACut.layerIds.back() == -1) {
+          foundAllLayerIds = false;
+          edm::LogWarning("Configuration")
+              << "Layer name '" << layerName << "' not found in the CAGraph. Please check CACuts parameter set.";
+        }
+      }
+
+      // Cut
+      thisCACut.cutValue = thisTriplet.cutValue;
+      thisCACut.hasValueByInnerLayerId = false;
+    }
+
+    setCutValuesByInnerLayerIds(caLayers);
+  }
+
+  void setCutValuesByInnerLayerIds(CAGraph &caLayers) {  //TODO: Is "const {" needed here?
+
+    for (auto &thisTriplet : valuesByLayerIds_) {
+      if (thisTriplet.hasValueByInnerLayerId)
+        continue;
+      auto it = std::find(thisTriplet.layerIds.begin(), thisTriplet.layerIds.end(), -1);
+      if (it != thisTriplet.layerIds.end())
+        continue;
+
+      std::cout << "(" << thisTriplet.layerIds[0] << ", " << thisTriplet.layerIds[1] << ", " << thisTriplet.layerIds[2]
```

Comment:
Please remove this cout (temporary debug?) or replace it with a LogWarning or similar thread safe messaging

Reply:
Removed, thanks!